### ISSUE TITLE
bug: pre-attention eviction produces NaN logits and perplexity when a chunk exceeds cache capacity

### ISSUE THREAD

**Issue by umran666** [umran666/UniqKache#59]
### Bug and location

[src/uniqkache/models/synthetic.py:275-296](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/models/synthetic.py#L275-L296) appends an entire query chunk before computing attention. `KVCache.append()` immediately enforces capacity at [src/uniqkache/cache/kv_cache.py:173-175](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/cache/kv_cache.py#L173-L175).

With a sliding window, no protected sinks, and a chunk longer than capacity, the surviving keys are all in the future for the chunk's earlier queries. Causal masking makes every score in those rows `-inf`; softmax returns NaN.

### Reproduction

```python
import torch
from uniqkache import KVCache, build_policy
from uniqkache.models.synthetic import build_model
from uniqkache.metrics.quality import random_token_ids
from uniqkache.bench.config import RunSpec
from uniqkache.bench.runner import run_spec

torch.set_num_threads(1)
model = build_model("tiny")
cache = KVCache(model.config.cache_config(capacity=4),
                policy=build_policy("sliding_window"))
with torch.no_grad():
    logits, _ = model.forward(random_token_ids(512, 16), cache)
print(logits.isnan().sum().item(), logits.numel())

outcome = run_spec(RunSpec(
    context_length=16, capacity=4, policy="sliding_window",
    attention_sinks=0, quality_chunk_size=16,
    max_new_tokens=2, device="cpu",
))
print(outcome.record.quality_value)
```

Observed: `6144 8192` and `nan`. The first 12 query positions have invalid logits; the final four remain finite. Default one-token quality chunks do not trigger this particular reproduction.

### Impact and expected behavior

Supported multi-token forward/chunked quality evaluation yields invalid logits and unusable perplexity. Eviction must preserve the keys required by each query until attention is computed, or the implementation must use a causally valid chunking/eviction schedule. Test chunks both below and above capacity with zero sinks.

### Verification

Reviewed commit: `0b878a1acc3f2a84ea0ff16194456b8ed10191e8`. Reproduced on Windows with Python 3.10 and PyTorch 2.6.0+cu124; CPU unless otherwise specified. Synthetic-model results below are correctness diagnostics, not language-model quality claims.
